clap_derive/src/derives/args.rs

```diff
@@ -361,7 +361,7 @@ pub fn gen_augment(
 
 fn gen_arg_enum_possible_values(ty: &Type) -> TokenStream {
```

Comment:
Do you plan to update `ArgEnum` in this PR or in another one?

Reply:
@epage I don't have a lot of experience in rust macros, but I'll have a look at it.

Reply:
Depending on what you are comfortable with, I can help mentor or go ahead and do it.  If you want real time access, feel free to let me know what platform works for you.  I'm on gitter, two Rust discord groups, and rust-lang's Zulip.  I can probably make something else work too.

This should be a relatively simple one,  I'd recommend it being a separate PR, regardless of who does it, just to make reviewing simple.
1. Update the `ArgEnum` trait in `src/derive.rs`
2. Update `clap_derive/src/derives/arg_enum.rs` for the changed trait.  Its mostly changing of a template or duplicating some templates, so hopefully it looks straighforward

We could stop there and make it a PR so we can focus on the trait design.   The next step is supporting the `hidden` and `about` attributes.  These mirror `Arg` attributes, so we can just leverage `Arg`s logic.  The key part from `clap_derive/src/derives/args.rs` is
```rust
                let name = attrs.cased_name();
                let methods = attrs.field_methods();

                Some(quote_spanned! { field.span()=>
                    let #app_var = #app_var.arg(
                        clap::Arg::new(#name)
                            #modifier
                            #methods
                    );
                })
```

When we create the `ArgValue`, we can call `attrs.field_methods` to convert attributes into calls and use the templating to insert it (`#methods`).

Of course, this is all hand wavy and I don't remember what limitations exist in the code (e.g. I think you can use "field_methods" despite this being methods for a variant).  `cargo expand` will be your friend through this.

Reply:
alright, would also prefer to do the derive seperatly.